hypothesis.py

```diff
@@ -86,75 +150,86 @@ def get_predictions(sample_data: pd.DataFrame,
+  aggregate_columns = [
       _TREE_CODE_COLUMN_NAME,
       _LONGITUDE_COLUMN_NAME,
       _LATITUDE_COLUMN_NAME,
-      _FRAUD_LABEL_COLUMN_NAME])[isotope_column_names]
-  predictions = pd.DataFrame({_TREE_CODE_COLUMN_NAME: [],
-                              _LONGITUDE_COLUMN_NAME: [],
-                              _LATITUDE_COLUMN_NAME: [],
-                              _FRAUD_LABEL_COLUMN_NAME: [],
-                              _FRAUD_P_VALUE_COLUMN_NAME: []})
+      _FRAUD_LABEL_COLUMN_NAME]
 
-  for group_key, isotope_values in sample_data:
-    if isotope_values.shape[0] <= 1:
-      continue
+  feature_columns = list(sample_data.columns.values)
+  for col in isotope_column_names:
+    feature_columns.remove(col)
 
-    p_values = []
-    for i, isotope_column_name in enumerate(isotope_column_names):
-      hypothesis_test = sample_ttest(longitude=group_key[1],
-                                     latitude=group_key[2],
-                                     isotope_values=isotope_values[isotope_column_name],
-                                     means_isoscape=means_isoscapes[i],
-                                     variances_isoscape=variances_isoscapes[i],
-                                     sample_size_per_location=sample_size_per_location,
-                                     p_value_target=None)
-      p_values.append(hypothesis_test.p_value)
-    combined_p_value = np.array(p_values).prod()
-    if np.isnan(combined_p_value):
-      continue
-
-    row = {_TREE_CODE_COLUMN_NAME: [group_key[0]],
-           _LONGITUDE_COLUMN_NAME: [group_key[1]],
-           _LATITUDE_COLUMN_NAME: [group_key[2]],
-           _FRAUD_LABEL_COLUMN_NAME: [group_key[3]],
-           _FRAUD_P_VALUE_COLUMN_NAME: [combined_p_value]}
-    predictions = pd.concat([predictions, pd.DataFrame(row)], ignore_index=True)
+  sample_data = dataset.preprocess_sample_data(
+    df=sample_data,
+    feature_columns=feature_columns,
+    label_columns=isotope_column_names,
+    aggregate_columns=aggregate_columns,
+    keep_grouping=True
+  )
 
-  return predictions
+  return get_predictions_grouped(
+    sample_data=sample_data,
+    isotope_means_column_names=[f"{col}_mean" for col in isotope_column_names],
+    isotope_variances_column_names=[f"{col}_variance" for col in isotope_column_names],
+    isotope_counts_column_names=[f"{col}_{dataset.SAMPLE_COUNT_COLUMN_NAME_SUFFIX}" for col in isotope_column_names],
+    means_isoscapes=means_isoscapes,
+    variances_isoscapes=variances_isoscapes,
+    sample_size_per_location=sample_size_per_location)
 
 def fraud_metrics(sample_data: pd.DataFrame,
                   isotope_column_names: list[str],
                   means_isoscapes: list[raster.AmazonGeoTiff],
                   variances_isoscapes: list[raster.AmazonGeoTiff],
                   sample_size_per_location: int,
-                  p_value_target: float):
+                  p_value_target: float,
+                  group_data: bool = True):
```

Comment:
is there any way to gell if the data is already grouped without passing an explicit flag?  Shouldnt there be statistical columns in that case?